### PR TITLE
feat: change password

### LINKED ISSUE #32: Implement change password functionality
It is a new feature and supports  https://github.com/verdaccio/verdaccio/issues/401.

### PR DESCRIPTION
**Pull request by juanpicado**
**Type:** feat

The following has been addressed in the PR:

*  There is a related issue?
*  Unit or Functional tests are included in the PR

**Description:**

Based on #36 

 Resolves #32

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -148,3 +148,49 @@ export function sanityCheck(
...
+ * @returns {string}
+ */
+export function changePasswordToHTPasswd(
+  body: string,
+  user: string,
+  passwd: string,
+  newPasswd: string
+): string {
+  if (crypt3) {
+    passwd = crypt3(passwd);
```
Comment (priscilawebdev) [inline src/utils.js]: According to FP Programming, we should not reassign the parameters ... but ... after reading this chapter of the book 'Functional Light...'  https://github.com/getify/Functional-Light-JS/blob/master/manuscript/ch5.md/#chapter-5-reducing-side-effects . This is something that should be analyzed. So...in this case...I think it's fine.

File: src/htpasswd.js [hunk 2]
```diff
@@ -204,4 +197,76 @@ export default class HTPasswd {
...
+
+      if (!err) {
+        locked = true;
+      }
+
+      if (err && err.code !== 'ENOENT') {
+        return cb(err);
+      }
+
+      let body = this._stringToUt8(res);
```
Comment (priscilawebdev) [inline src/htpasswd.js]: I'd go for a const...because body is not being reassigned

File: src/utils.js [hunk 3]
```diff
@@ -148,3 +148,49 @@ export function sanityCheck(
...
+    newPasswd = crypt3(newPasswd);
+  } else {
+    passwd = getCryptoPassword(passwd);
+    newPasswd = getCryptoPassword(newPasswd);
+  }
+
+  let lines = body.split('\n');
+  lines = lines.map(line => {
+    let [username, password] = line.split(':', 3);
```
Comment (priscilawebdev) [inline src/utils.js]: const?

File: src/utils.js [hunk 4]
```diff
@@ -148,3 +148,49 @@ export function sanityCheck(
...
+    if (username == user) {
```
Comment (priscilawebdev) [inline src/utils.js]: Won't they be of the same type?

File: src/utils.js [hunk 5]
```diff
@@ -148,3 +148,49 @@ export function sanityCheck(
...
+      if (password == passwd) {
```
Comment (priscilawebdev) [inline src/utils.js]: Won't they be of the same type?

File: src/utils.js [hunk 6]
```diff
@@ -148,3 +148,49 @@ export function sanityCheck(
...
+ * @param {string} passwd
+ * @param {string} newPasswd
+ * @returns {string}
+ */
+export function changePasswordToHTPasswd(
+  body: string,
+  user: string,
+  passwd: string,
+  newPasswd: string
+): string {
```
Comment (priscilawebdev) [inline src/utils.js]: Shouldn't it return an object?

File: src/utils.js (hunk 5 above)
Comment (juanpicado) [reply src/utils.js]: good one, I wonder why eslint does not catch this.

File: src/utils.js (hunk 6 above)
Comment (juanpicado) [reply src/utils.js]: It returns a string, the whole HTPasswd body content as string.

**Comment by juanpicado** [comment]
It decreases a bit the coverage, but is due a check is hard to mock, whether `crypt3` exist or not.

File: src/utils.js (hunk 6 above)
Comment (priscilawebdev) [reply src/utils.js]: ok.. 👍

**Pull request merged**